File: L2JBrasil_DP/data/jscript/ai/individual/nurka.py
import sys
from com.it.br.gameserver.ai import CtrlIntention
from com.it.br.gameserver.instancemanager.clanhallsiege import FortResistSiege
from com.it.br.gameserver.model.quest import State
from com.it.br.gameserver.model.quest import QuestState
from com.it.br.gameserver.model.quest.jython import QuestJython as JQuest
from com.it.br.gameserver.instancemanager import ClanHallManager
from com.it.br.util import Rnd
from java.lang import System

NURKA = 35368
MESSENGER = 35382
CLANLEADERS = []

class Nurka(JQuest):

 def __init__(self,id,name,descr): JQuest.__init__(self,id,name,descr)

 def onTalk (self,npc,player):
   global CLANLEADERS
   npcId = npc.getNpcId()
   if npcId == MESSENGER :
     for clname in CLANLEADERS:
       if player.getName() == clname :
         return "<html><body>You already registered!</body></html>"
     if FortResistSiege.getInstance().Conditions(player) :
       CLANLEADERS.append(player.getName())
       return "<html><body>You have successful registered on a battle</body></html>"
     else:
       return "<html><body>Condition are not allow to do that!</body></html>"
   return
 
 def onAttack (self,npc,player,damage,isPet):
   global CLANLEADERS
   for clname in CLANLEADERS:
     if clname <> None :
       if player.getClan().getLeader().getName() == clname :
         FortResistSiege.getInstance().addSiegeDamage(player.getClan(),damage)
   return

 def onKill(self,npc,player,isPet):
   FortResistSiege.getInstance().CaptureFinish()
   return

QUEST = Nurka(-1, "nurka", "ai")
CREATED = State('Start', QUEST)
QUEST.setInitialState(CREATED)

QUEST.addTalkId(MESSENGER)
QUEST.addStartNpc(MESSENGER)

QUEST.addAttackId(NURKA)
QUEST.addKillId(NURKA)

File: L2JBrasil_CORE/java/com/it/br/gameserver/ai/special/individual/Nurka.java
package com.it.br.gameserver.ai.special.individual;

import com.it.br.gameserver.instancemanager.clanhallsiege.FortResistSiege;
import com.it.br.gameserver.model.actor.instance.L2NpcInstance;
import com.it.br.gameserver.model.actor.instance.L2PcInstance;
import com.it.br.gameserver.model.quest.Quest;
import com.it.br.gameserver.model.quest.State;

import java.util.ArrayList;
import java.util.List;

public class Nurka extends Quest implements Runnable
{
	private static final int NURKA = 35368;
	private static final int MESSENGER = 35382;
	private static List<String> CLAN_LEADERS = new ArrayList<>();

	public Nurka(int questId, String name, String descr)
	{
		super(questId, name, descr);
		State CREATED = new State("Start", this);
		this.setInitialState(CREATED);
				
		addTalkId(MESSENGER);
		addStartNpc(MESSENGER);
		addAttackId(NURKA);
		addKillId(NURKA);
	}
	
	@Override
	public String onTalk(L2NpcInstance npc, L2PcInstance player)
	{
		if(npc.getNpcId() == MESSENGER)
		{
			if(player != null && CLAN_LEADERS.contains(player.getName()))
				return "<html><body>Messenger:<br>You already registered!</body></html>";
			else if(FortResistSiege.Conditions(player))
			{
				CLAN_LEADERS.add(player.getName());
				return "<html><body>Messenger:<br>You have successful registered on a siege!</body></html>";
			}
			else
				return "<html><body>Messenger:<br>Condition are not allow to do that!</body></html>";		
		}

		return super.onTalk(npc, player);
	}

	@Override
	public String onAttack(L2NpcInstance npc, L2PcInstance attacker, int damage, boolean isPet)
	{
		if(attacker != null && npc.getNpcId() == NURKA && CLAN_LEADERS.contains(attacker.getName()));
			FortResistSiege.getInstance().addSiegeDamage(attacker.getClan(),damage);

		return super.onAttack(npc, attacker, damage, isPet);
	}

	@Override
	public String onKill(L2NpcInstance npc, L2PcInstance killer, boolean isPet)
	{
		FortResistSiege.getInstance().CaptureFinish();
			return super.onKill(npc, killer, isPet);
	}

    public static void main(String[] args)
    {
    	new Nurka(-1,"Nurka","ai");
    }

	@Override
	public void run()
	{}
}